plugins/base.py

```diff
@@ -83,50 +126,116 @@ def ensure_authorization(self):
         self.cookies.save()
 
     @abstractmethod
-    def get_server_name(self):
+    def can_process_torrent(self, torrent):
+        # type: (Torrent) -> bool
+        """
+        Check if torrent could be processed with this plugin
+        :param torrent: Torrent object
+        :return: True if torrent could be processed with this plugin
+        """
         pass
 
     @abstractmethod
     def is_authorized(self, opener):
+        # type: (urllib2.OpenerDirector) -> bool
+        """
+        Check authorization status(usually by trying to load resource not
+        allowed for guests)
+        :param opener: URL opener for network operations
+        :return: True if authorised
+        """
         return False
 
     @abstractmethod
     def authorize(self, opener):
+        # type: (urllib2.OpenerDirector) -> None
+        """
+        Perform authorization at torrent tracker
+        :param opener: URL opener for network operations
+        """
         pass
 
     @abstractmethod
-    def get_topic_url(self, torrent_id):
+    def get_topic_url(self, torrent):
+        # type: (Torrent) -> str
+        """
+        Return absolute URL of torrent(topic)
+        :param torrent: Torrent object
+        :return: URL string
+        """
         pass
 
     @abstractmethod
-    def load_description(self, torrent_id):
+    def load_description(self, torrent):
+        # type: (Torrent) -> str
+        """
+        Load torrent description(title of a topic)
+        :param torrent: Torrent object
+        :return: Description string
+        """
         pass
 
     @abstractmethod
-    def load_torrent(self, torrent_id):
+    def load_torrent_data(self, torrent):
+        # type: (Torrent) -> str
+        """
+        Load torrent file
+        :param torrent: Torrent object
+        :return: Data of torrent file in string representation and
+            updated torrent state object
+        """
         pass
 
     @abstractmethod
-    def find_torrents(self, query, type="movie"):
+    def find_torrents(self, query, category="new-movie"):
+        """
+        Return torrents matching query and with specific type.
+        :param query: Query string for search
+        :param category: Category of torrent (see server plugins)
+        :return Generator with matching torrent objects
+        """
         pass
 
 
 class OnStartPlugin(BasePlugin):
+    """
+    Base class for plugins that implement on-start logic like
+    something that is should be done before torrents checking process.
```

Comment:
Base class for plugins that implement on-start logic. When registered, these plugins will be triggered at the very beginning of the main process.